auxiliary/compileAMICIDependencies.m

```diff
@@ -0,0 +1,343 @@
+function [objectsstr, includesstr] = compileAMICIDependencies(wrap_path, o_suffix, COPT, DEBUG)
+    %COMPILEAMICIDEPENDENCIES Compiles Sundials and SuiteSparse libraries required by AMICI
+
+    if(strcmp(DEBUG, ''))
+        objectFolder = fullfile(wrap_path,'models',mexext, 'release');
+    else
+        objectFolder = fullfile(wrap_path,'models',mexext, 'debug');
+    end
+
+    sundials_path = fullfile(wrap_path,'sundials');
+    sundials_ver = '2.7.0';
+    
+    ssparse_path = fullfile(wrap_path,'SuiteSparse');
+    ssparse_ver = '4.5.3';
+    
+    lapack_path = fullfile(wrap_path,'lapack-3.5.0'); % currently not used, lapack implementation still needs to be done
+    lapack_ver = '3.5.0';
+    
+    % compile directory
+    if(~exist(objectFolder, 'dir'))
+        mkdir(objectFolder)
+    end
+    
+    version_file = fullfile(objectFolder, 'versions.txt');
+    [del_sundials, del_ssparse, del_lapack] = checkVersions(version_file, sundials_ver, ssparse_ver, lapack_ver);
+    
+    % assemble objectsstr
+    objectsstr = '';
+    objects_sundials = getObjectsSundials(o_suffix);
+    for j=1:length(objects_sundials)
+        objectsstr = strcat(objectsstr,' "',fullfile(objectFolder,objects_sundials{j}),'"');
+    end
+    
+    objects_ssparse = getObjectsSSparse(o_suffix);
+    for j=1:length(objects_ssparse)
+        objectsstr = strcat(objectsstr,' "',fullfile(objectFolder,objects_ssparse{j}),'"');
+    end
+        
+    includesstr = '';
+    includesstr = strcat(includesstr,' -I"', fullfile(sundials_path, 'include'), '"');
+    includesstr = strcat(includesstr,' -I"', fullfile(sundials_path, 'src','cvodes'), '"');
+    includesstr = strcat(includesstr,' -I"', fullfile(wrap_path), '"');
+    includesstr = strcat(includesstr,' -I"', fullfile(wrap_path, 'src'), '"');
+    includesstr = strcat(includesstr,' -I"', fullfile(wrap_path, 'include'), '"');
+    includesstr = strcat(includesstr,' -I"', fullfile(ssparse_path, 'KLU','Include'), '"');
+    includesstr = strcat(includesstr,' -I"', fullfile(ssparse_path, 'AMD','Include'), '"');
+    includesstr = strcat(includesstr,' -I"', fullfile(ssparse_path, 'COLAMD','Include'), '"');
+    includesstr = strcat(includesstr,' -I"', fullfile(ssparse_path, 'BTF','Include'), '"');
+    includesstr = strcat(includesstr,' -I"', fullfile(ssparse_path, 'SuiteSparse_config'), '"');
+    
+    % compile all the sundials objects if we haven't done so yet
+    sources_sundials = getSourcesSundials();
+    for j=1:length(sources_sundials)
+        if(~exist(fullfile(objectFolder,objects_sundials{j}), 'file') || del_sundials)
```

Comment:
can we also compile all the objects in a single call to mex as we do it for the model specific files?